ceph.spec.in

```diff
@@ -1120,49 +1223,71 @@ fi
-/sbin/ldconfig
+test -n "$FIRST_ARG" || FIRST_ARG=$1
 %if 0%{?suse_version}
-  # explicit systemctl daemon-reload (that's the only relevant bit of
-  # service_add_post; the rest is all sysvinit --> systemd migration which
-  # isn't applicable in this context (see above comment).
-  /usr/bin/systemctl daemon-reload >/dev/null 2>&1 || :
+%service_add_pre ceph-radosgw.target
+%endif
+%if 0%{?fedora} || 0%{?rhel}
+/usr/bin/systemctl daemon-reload >/dev/null 2>&1 || :
+if [ $FIRST_ARG -eq 1 ]; then
+  /usr/bin/systemctl preset ceph-radosgw.target >/dev/null 2>&1 || :
+fi
 %endif
 
 %preun radosgw
-%if 0%{?_with_systemd}
-  # Disable and stop on removal.
-  if [ $1 = 0 ] ; then
-    SERVICE_LIST=$(systemctl | grep -E '^ceph-radosgw@'  | cut -d' ' -f1)
-    if [ -n "$SERVICE_LIST" ]; then
-      for SERVICE in $SERVICE_LIST; do
-        /usr/bin/systemctl --no-reload disable $SERVICE > /dev/null 2>&1 || :
-        /usr/bin/systemctl stop $SERVICE > /dev/null 2>&1 || :
-      done
-    fi
-  fi
+test -n "$FIRST_ARG" || FIRST_ARG=$1
+%if 0%{?suse_version}
+%service_del_preun ceph-radosgw.target
+%endif
+%if 0%{?fedora} || 0%{?rhel}
+if [ $FIRST_ARG -eq 0 ]; then
+  /usr/bin/systemctl --no-reload disable ceph-radosgw.target > /dev/null 2>&1 || :
+  /usr/bin/systemctl stop ceph-radosgw.target > /dev/null 2>&1 || :
+fi
 %endif
+if [ $FIRST_ARG -eq 0 ]; then
+  SERVICE_LIST=$(systemctl | grep -E '^ceph-radosgw@' | cut -d' ' -f1)
+  if [ -n "$SERVICE_LIST" ]; then
+    for SERVICE in $SERVICE_LIST; do
+      /usr/bin/systemctl --no-reload disable $SERVICE > /dev/null 2>&1 || :
+      /usr/bin/systemctl stop $SERVICE > /dev/null 2>&1 || :
+    done
+  fi
+fi
 
 %postun radosgw
-/sbin/ldconfig
-%if 0%{?_with_systemd}
-  if [ $1 = 1 ] ; then
-    # Restart on upgrade, but only if "CEPH_AUTO_RESTART_ON_UPGRADE" is set to
-    # "yes". In any case: if units are not running, do not touch them.
-    SYSCONF_CEPH=/etc/sysconfig/ceph
-    if [ -f $SYSCONF_CEPH -a -r $SYSCONF_CEPH ] ; then
-      source $SYSCONF_CEPH
-    fi
-    if [ "X$CEPH_AUTO_RESTART_ON_UPGRADE" = "Xyes" ] ; then
-      SERVICE_LIST=$(systemctl | grep -E '^ceph-radosgw@'  | cut -d' ' -f1)
-      if [ -n "$SERVICE_LIST" ]; then
-        for SERVICE in $SERVICE_LIST; do
-          /usr/bin/systemctl try-restart $SERVICE > /dev/null 2>&1 || :
-        done
-      fi
+test -n "$FIRST_ARG" || FIRST_ARG=$1
+%if 0%{?suse_version}
+DISABLE_RESTART_ON_UPDATE="yes"
+%service_del_postun ceph-radosgw.target
+%endif
+%if 0%{?fedora} || 0%{?rhel}
+if [ $FIRST_ARG -eq 0 ] ; then
+  /usr/bin/systemctl daemon-reload >/dev/null 2>&1 || :
+fi
+%endif
+if [ $FIRST_ARG -ge 1 ] ; then
+  # Restart on upgrade, but only if "CEPH_AUTO_RESTART_ON_UPGRADE" is set to
+  # "yes". In any case: if units are not running, do not touch them.
+  SYSCONF_CEPH=/etc/sysconfig/ceph
+  if [ -f $SYSCONF_CEPH -a -r $SYSCONF_CEPH ] ; then
+    source $SYSCONF_CEPH
+  fi
+  if [ "X$CEPH_AUTO_RESTART_ON_UPGRADE" == "Xyes" ] ; then
```

Comment:
Strictly, should be a single = for POSIX compliance (some non-bash shells don't like ==)